modules/flann/include/opencv2/flann/dist.h

```diff
@@ -683,6 +683,8 @@ struct Hamming2
     template <typename Iterator1, typename Iterator2>
     ResultType operator()(const Iterator1 a, const Iterator2 b, size_t size, ResultType /*worst_dist*/ = -1) const
     {
+        CV_DbgAssert(!(size % long_word_size_) && "vectors size must be multiple of long words size (i.e. 8)");
```

Comment:
Is it frequently used function (for example called from some loop)? Is overhead of this check measurable?
If not then my suggestion is to use `CV_Assert()` instead to keep this check in Release builds too (almost all binary packages + python bindings).
Or we need to add an upper level validation check for that (which is called once to reduce overhead).